### ISSUE TITLE
No way to pass DataBindingComponent to DataBindingEpoxyModel

### ISSUE THREAD

**Issue by ubuntudroid** [airbnb/epoxy#711]
We frequently use `DataBindingComponent` to provide Glide instances tied to the current fragment to our binding adapters.

However when using Epoxy we can't pass this component to our autogenerated RecyclerView models inheriting `DataBindingEpoxyModel`.

It would be really great if we would be able to pass the component to the generated `*ModelBuilder` which would in turn pass it to the built model which then uses it in `buildView()` during layout inflation.

**Comment by kalyandechiraju** [comment]
Yes, this feature is missing. However, we can implement this by extending DataBindingEpoxyModel.
Here is the sample:

```kotlin
class ItemModel(
    private val dataBindingComponent: DataBindingComponent,
    private val onItemClickListener: ((String?) -> Unit)?, 
    // and any other params
) : DataBindingEpoxyModel() {

	override fun buildView(parent: ViewGroup): View {
		val binding = DataBindingUtil.inflate<ViewBinding>(
                        LayoutInflater.from(parent.context),
			R.layout.item_layout,
			parent,
			false,
			dataBindingComponent
		)
		val view = binding.root
		view.tag = binding
		return view
	}

        // We don't even need this, if you make the model abstract
        override fun getDefaultLayout() = R.layout.item_layout

        override fun setDataBindingVariables(binding: ViewDataBinding?) {
            // Add your binding logic here
        }
}
```

While adding to controller:
```kotlin
ItemModel(dataBindingComponent, itemClickListener).apply {
    id(someId)
}.addTo(controller: this)
```

**Comment by elihart** [comment]
We don't use databinding, so I'm not sure what the best solution would be.

If you have an idea for how to change the generated databinding models feel free to put up a PR.

The approach we take with GLide is to not wory about passing anything to the view and instead get the global glide instance from the view context.. The view has a lifecycle tied to the Recyclerview - so the view will be detached from the window when it is recycled, or when the fragment view is destroyed (as long as the recyclerview is configured to clear then).

Image requests can be canceled then (caveat, they should also be restarted in onAttachedToWindow since detach from window is not symmetric with view binding).

This works better with model views since they have a @OnViewRecycled callback - it's possible that databinding generated models could set up some sort of callback to clean up images.

Or from your epoxy controller you can register an `unbind` listener on the model and clear the image requests from that.
